src/engine/order.ts

```diff
@@ -47,53 +41,58 @@ export class OrderEngine {
+    const rawPrice = quote ? quote : await this.quoteEngine.getBuyPrice()
+    const price = this.exchangeProvider.priceToPrecision(exchange, symbol, rawPrice)
 
     const purchasingPower = this.exchangeProvider.priceToPrecision(exchange, symbol, this.wallet.currency * strength)
     const amount = this.exchangeProvider.amountToPrecision((purchasingPower / price) * 0.995)
-    // console.log({ price, amount, purchasingPower })
-    // process.exit()
 
     // @todo(notVitaliy): Add support for market
     const orderOpts: OrderOpts = { symbol, price, amount, type: 'limit', side: 'buy' }
+    const adjustment: Adjustment = { asset: 0, currency: -(amount * price), type: 'newOrder' }
 
-    const order = await this.placeOrder(orderOpts)
+    const order = await this.placeOrder(orderOpts, adjustment)
     if (!order) return
 
-    const { id } = order
-    this.emitWalletAdjusment({ asset: 0, currency: -(amount * price), type: 'newOrder' })
-
-    await this.checkOrder(id)
+    await this.checkOrder(order.id)
   }
 
   public async executeSell(quote?: number, strength = 1) {
     const { exchange, symbol } = this.opts
-    const price = this.exchangeProvider.priceToPrecision(exchange, symbol, quote ? quote : await this.quoteEngine.getSellPrice())
+    const rawPrice = quote ? quote : await this.quoteEngine.getSellPrice()
+    const price = this.exchangeProvider.priceToPrecision(exchange, symbol, rawPrice)
 
     const amount = this.exchangeProvider.amountToPrecision(this.wallet.asset * strength)
 
     // @todo(notVitaliy): Add support for market
     const orderOpts: OrderOpts = { symbol, price, amount, type: 'limit', side: 'sell' }
+    const adjustment: Adjustment = { asset: -amount, currency: 0, type: 'newOrder' }
 
-    const order = await this.placeOrder(orderOpts)
+    const order = await this.placeOrder(orderOpts, adjustment)
     if (!order) return
 
-    const { id } = order
-    this.emitWalletAdjusment({ asset: -amount, currency: 0, type: 'newOrder' })
-
-    await this.checkOrder(id)
+    await this.checkOrder(order.id)
   }
 
-  private async placeOrder(orderOpts: OrderOpts) {
+  private async placeOrder(orderOpts: OrderOpts, adjustment: Adjustment) {
     logger.info('Placing order:', orderOpts)
 
-    const { exchange } = this.opts
-    const order = await this.exchangeProvider.placeOrder(exchange, orderOpts)
-    if (!order || !order.id) return false
+    try {
+      const { exchange } = this.opts
+      const order = await this.exchangeProvider.placeOrder(exchange, orderOpts)
+      if (!order || !order.id) return false
 
-    await this.orderStore.newOrder(order)
+      // Great Success! Decrement wallet
+      this.emitWalletAdjustment(adjustment)
+      await this.orderStore.newOrder(order)
 
-    return order
+      return order
+    } catch (e) {
+      // No Monies!! Decrement wallet
+      if (e instanceof InsufficientFunds) this.emitWalletAdjustment(adjustment)
+      logger.error(e.message)
```

Comment:
i would love to see more context on this log message. 
As user you will now see here just the message of the Exchange (if there is any).
You could add some Context like `Order was not placed, exchange returned error: ${e.message}`